### DISCUSSION TITLE
How to to debug an application?

### DISCUSSION THREAD

**Discussion by jpsacha** [scala-native/scala-native#3310] [Q&A]
I am using IntelliJ to work in an application ((https://github.com/ij-plugins/ijp-imagej-launcher). I can debug or run an application as long as it is not attempting to use native code (I can run it from SBT). 

How to debug an application when it is using native calls (@extern) assuming that I do not need to step into a native call?

**Comment by armanbilge** [comment]
Personally I use `println(...)`s and minimize code to track down bugs 😅 

I guess it should be possible to use [lldb](https://lldb.llvm.org/), esp. with upcoming improvements like https://github.com/scala-native/scala-native/pull/2869#issuecomment-1573500134, as well as address sanitization as explained in https://github.com/scala-native/scala-native/issues/1918#issuecomment-849253272.

**Comment by WojciechMazur** [comment]
When struggling with undefined behavior lldb running in disassembly mode is probably your best friend...at least until we would be able to map Scala sources in lldb. Unfortunately, whole Scala tooling is oriented around JVM, so having a decent debugging experience is a really long journey.
